bin/faucet/src/client.rs

```diff
@@ -0,0 +1,353 @@
+        &mut self,
+        target_account_id: AccountId,
+        is_private_note: bool,
+        asset_amount: u64,
+    ) -> Result<(ExecutedTransaction, Note), FaucetError> {
+        let asset = FungibleAsset::new(self.id, asset_amount)
+            .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
+
+        let note_type = if is_private_note {
+            NoteType::OffChain
+        } else {
+            NoteType::Public
+        };
+
+        let output_note =
+            create_p2id_note(self.id, target_account_id, vec![asset.into()], note_type, self.rng)
+                .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
+
+        let recipient = output_note
+            .recipient()
+            .digest()
+            .iter()
+            .map(|x| x.as_int().to_string())
+            .collect::<Vec<_>>()
+            .join(".");
+
+        let tag = output_note.metadata().tag().inner();
+
+        let script = ProgramAst::parse(
+            &DISTRIBUTE_FUNGIBLE_ASSET_SCRIPT
+                .replace("{recipient}", &recipient)
+                .replace("{note_type}", &Felt::new(note_type as u64).to_string())
+                .replace("{tag}", &Felt::new(tag.into()).to_string())
+                .replace("{amount}", &Felt::new(asset.amount()).to_string()),
+        )
+        .expect("shipped MASM is well-formed");
+
+        let script = self.executor.compile_tx_script(script, vec![], vec![]).map_err(|err| {
+            FaucetError::InternalServerError(format!("Failed to compile script: {}", err))
+        })?;
+
+        let mut transaction_args = TransactionArgs::new(Some(script), None, AdviceMap::new());
+        transaction_args.extend_expected_output_notes(vec![output_note.clone()]);
+
+        self.executor
+            .load_account(self.id)
+            .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
+
+        let executed_tx =
+            self.executor.execute_transaction(self.id, 0, &[], transaction_args).unwrap();
+
+        Ok((executed_tx, output_note))
+    }
+
+    pub async fn prove_and_submit_transaction(
+        &mut self,
+        executed_tx: ExecutedTransaction,
+    ) -> Result<(), FaucetError> {
+        let transaction_prover = TransactionProver::new(ProvingOptions::default());
+
+        let delta = executed_tx.account_delta().clone();
+
+        let proven_transaction =
+            transaction_prover.prove_transaction(executed_tx).map_err(|err| {
+                FaucetError::InternalServerError(format!("Failed to prove transaction: {}", err))
+            })?;
+
+        let request = SubmitProvenTransactionRequest {
+            transaction: proven_transaction.to_bytes(),
+        };
+
+        self.rpc_api
+            .submit_proven_transaction(request)
+            .await
+            .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
+
+        self.data_store.update_faucet_account(&delta).unwrap();
```

Comment:
This should probably return an error instead of panicking

Reply:
Done.